Tests/NIOTests/UniversalBootstrapSupportTest.swift

```diff
@@ -135,4 +135,98 @@ class UniversalBootstrapSupportTest: XCTestCase {
+    
+    func testBootstrapOverrideOfShortcutOptions() {
+        final class FakeBootstrap : NIOClientTCPBootstrapProtocol {
+            func channelInitializer(_ handler: @escaping (Channel) -> EventLoopFuture<Void>) -> Self {
+                fatalError("Not implemented")
+            }
+            
+            func protocolHandlers(_ handlers: @escaping () -> [ChannelHandler]) -> Self {
+                fatalError("Not implemented")
+            }
+            
+            var regularOptionsSeen = false
+            func channelOption<Option>(_ option: Option, value: Option.Value) -> Self where Option : ChannelOption {
+                regularOptionsSeen = true
+                return self
+            }
+            
+            var convenienceOptionConsumed = false
+            func _applyChannelConvenienceOptions(_ options: inout ChannelOptions.TCPConvenienceOptions) -> FakeBootstrap {
+                if options.consumeAllowLocalEndpointReuse().isSet {
+                    convenienceOptionConsumed = true
+                }
+                return self
+            }
+            
+            func connectTimeout(_ timeout: TimeAmount) -> Self {
+                fatalError("Not implemented")
+            }
+            
+            func connect(host: String, port: Int) -> EventLoopFuture<Channel> {
+                fatalError("Not implemented")
+            }
+            
+            func connect(to address: SocketAddress) -> EventLoopFuture<Channel> {
+                fatalError("Not implemented")
+            }
+            
+            func connect(unixDomainSocketPath: String) -> EventLoopFuture<Channel> {
+                fatalError("Not implemented")
+            }
+        }
+        
+        // Check consumption works.
+        let consumingFake = FakeBootstrap()
+        _ = NIOClientTCPBootstrap(consumingFake, tls: NIOInsecureNoTLS())
+            .channelConvenienceOptions([.allowLocalEndpointReuse])
+        XCTAssertTrue(consumingFake.convenienceOptionConsumed)
+        XCTAssertFalse(consumingFake.regularOptionsSeen)
+        
+        // Check default behaviour works.
+        let nonConsumingFake = FakeBootstrap()
+        _ = NIOClientTCPBootstrap(nonConsumingFake, tls: NIOInsecureNoTLS())
+            .channelConvenienceOptions([.allowRemoteHalfClosure])
+        XCTAssertFalse(nonConsumingFake.convenienceOptionConsumed)
+        XCTAssertTrue(nonConsumingFake.regularOptionsSeen)
+        
+        // Both at once.
+        let bothFake = FakeBootstrap()
+        _ = NIOClientTCPBootstrap(bothFake, tls: NIOInsecureNoTLS())
+            .channelConvenienceOptions([.allowRemoteHalfClosure, .allowLocalEndpointReuse])
+        XCTAssertTrue(bothFake.convenienceOptionConsumed)
+        XCTAssertTrue(bothFake.regularOptionsSeen)
+    }
+}
+
+// Prove we've not broken implementors of NIOClientTCPBootstrapProtocol by adding a new method.
+private class UniversalWithoutNewMethods : NIOClientTCPBootstrapProtocol {
```

Comment:
@weissi - Class missing _applyChannelConvenienceOptions function to prove the API change will not break anyone.

Reply:
@Lukasa I'm guessing that you'll eventually merge this PR as your approval is still outstanding so I guess this is most relevant to you, this is the proof that we didn't actually break API despite what api-digester is saying.